instantsearch/src/main/java/com/algolia/instantsearch/model/SearchResults.java

```diff
@@ -27,6 +29,8 @@
     public Map<String, List<FacetValue>> disjunctiveFacets;
 
     // Mandatory attributes
+    /** Declarations of all hierarchical facets. */
+    private Map<String, List<String>> hierarchicalFacetDeclarations = new HashMap<>();
```

Comment:
@XiaowenVC, is there a need for requiring the developer to explicitly list all levels? 
- It could be pretty cumbersome to write for customers with a deep hierarchy
- It would require updating the code when the index is updated with a new level (e.g. if one adds `lvl3` in your tested example)
- InstantSearch.js' `hierarchicalMenu` widget only requires a `attributes` parameter and parses each attribute's hierarchy until the last level, which seems to provide a better (simpler) DX

Is there any reason not to go with the same approach? 🤔

Reply:
Just like on JS SDK, we leave the client to explicitly list all levels, in case they do not have a same naming rule for different levels. (ex: category.first, category.second...).

For JS SDK, "hierarhical menu" is a UI widget. So we just need to provide attributes, then it can display all values. But in Android SDK, "hierarchical facet" provides only data, if we have multiple hierarchical facets (ex: category, material), we should give each of them a name, in order to get the value by code `results.getHierarchicalFacet("category")` and `results.getHierarchicalFacet("material")`.